src/transformStateWithClones.js

```diff
@@ -6,8 +6,34 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  let copy = {};
+
+  Object.assign(copy, state);
+
+  const result = [];
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(copy, action.extraData);
+      result.push({ ...copy });
+    }
+
+    if (action.type === 'removeProperties') {
+      for (const key of action.keysToRemove) {
+        delete copy[key];
+      }
+      result.push({ ...copy });
+    }
+
+    if (action.type === 'clear') {
+      copy = {};
+      result.push({ ...copy });
```

Comment:
Pushing a copy of the object to the result array is repeated in each condition. According to the checklist, this should be done once after processing the current action, outside of the switch block.